src/state/presentationManagerImpl.ts

```diff
@@ -93,9 +96,12 @@ export class PresentationManagerImpl implements PresentationManager {
       .subscribe(({ senderSessionId, content }) => {
         if (content.view) {
           this.setCurrentPresenter(senderSessionId);
+          this.setEditMode(content.view.isEditMode);
+          content.view.isEditMode && whiteboardInstance.clearUndoManager();
```

Comment:
```suggestion
          if (content.view.isEditMode) {
            whiteboardInstance.clearUndoManager();
          }
```

Reply:
Note that this message can be received multiple times because it is broadcasted whenever another user opens or closes the widget. This could unintuitively reset the undo stack while the edit mode is on. Maybe we could only execute it when either the slideid changes or when the editmode is enabled.